src/main/java/com/karlofduty/NoVoidLoss/DeathListener.java

```diff
@@ -1,61 +1,21 @@
 package com.karlofduty.NoVoidLoss;
 
-import org.bukkit.Material;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
-import org.bukkit.event.entity.EntityDamageEvent;
 import org.bukkit.event.entity.PlayerDeathEvent;
-import org.bukkit.event.player.PlayerRespawnEvent;
-import org.bukkit.inventory.ItemStack;
-
-import java.util.Map;
 
 public class DeathListener implements Listener
 {
-    private NoVoidLoss plugin;
-
-    public DeathListener(final NoVoidLoss plugin)
-    {
-        this.plugin = plugin;
-    }
-
-    @EventHandler(priority = EventPriority.HIGHEST)
-    public void onEvent(final PlayerDeathEvent event)
-    {
-        boolean restoreAllItems = false;
-        try {
-            restoreAllItems = event.getEntity().getLastDamageCause().getCause() == EntityDamageEvent.DamageCause.VOID
-                           || event.getEntity().getLastDamageCause().getCause() == EntityDamageEvent.DamageCause.SUFFOCATION;
-        }
-        catch (NullPointerException ignored) { }
-
-        final ItemEntry entry = new ItemEntry();
-        for (int i = 0; i < event.getEntity().getInventory().getSize(); ++i)
-        {
-            final ItemStack stack = event.getEntity().getInventory().getItem(i);
-            if (stack != null && !stack.getType().equals(Material.AIR) && (stack.getType().equals(Material.PLAYER_HEAD) || restoreAllItems))
```

Comment:
Also the rewrite loses the player head protection

Reply:
Any item (including player heads) gets kept.

Reply:
Players should only keep other items when they die from the above two reasons.